python/src/aiconfig/editor/client/src/components/prompt/prompt_input/PromptInputRenderer.tsx

```diff
@@ -91,14 +91,20 @@ export default memo(function PromptInputRenderer({
     </Flex>
   );
 
+  const runPrompt = useCallback(async () => {
+    if (!(isRunButtonDisabled || isRunning)) {
```

Comment:
Ok, just to confirm, is the only case where this is possible to enter here is if the prompt (or another prompt) is running and as it's running, user decides to change the input? Otherwise for read-only mode they won't be able to change this anyways right

Reply:
Also nit: I think it's slightly easier to exit early on the true condition
```
if (isRunButtonDisabled || isRunning) {
  // Do not let us invoke run operation when we're supposed to
  return;
}
await onRunPrompt();
```

Reply:
Correct